File: src/ast/FunctionParameter.java
package ast;

import java.io.StringWriter;

public class FunctionParameter extends BaseNode {
    private String _number;

    public FunctionParameter(String number) {
        super(NodeType.FunctionParameter);
        _number = number;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("fp ");

        if (_number != null) {
            writer.write(_number);
        }
    }
}

File: src/de/fearlesstobi/demangler/ast/FunctionParameter.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class FunctionParameter(private val number: String?) : BaseNode(NodeType.FunctionParameter) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("fp ")
        if (number != null) {
            writer.write(number)
        }
    }

}